packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -32,17 +62,10 @@ function computeSelfTime(span: Span, allSpans: Span[]): number {
   // We should've ended up with length 9-4=5, but we got 3.
   // To work around that, we multiply start/end times by 10 and subtract one from the end.
   // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
-  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
-  // Only keep CHILD_OF spans. FOLLOWS_FROM spans do not block the parent.
-  const children = allSpans.filter(
-    each =>
-      span.childSpanIds.includes(each.spanID) &&
-      each.references[0].spanID === span.spanID &&
-      each.references[0].refType === 'CHILD_OF'
+  if (!span.hasChildren) return span.duration;
+  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1).subtract(
+    getChildOfDrange(span.spanID, allSpans)
```

Comment:
also unnecessary change, can keep the original logic
```suggestion
  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
  const children = getChildOfSpans(span.spanID, allSpans);
  children.forEach(child => {
    spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
```
it will also like be faster because it will maintain a single array of possible fragmented ranges, whereas in the new code a new fragmented array is created from children and then the parent array is possible transformed to fragmented via subtraction. But mainly I don't like changing working code without reason.